open-edge-platform/training_extensions
# Support multiple python versions up to 3.10

### Summary

* Support Python3.8/9/10 via tox
  - `pre-commit-all-py{38,39,310}`
  - `pre-merge-{all,ano,cls,det,seg,act}-py{38,39,310}`
* Keeping `pre-commit` & `pre-merge` and others for backward compatibility
  - Defaults to all-py38 for now



### How to test

* Specific Python version should be installed in your system
```console
(base) $ conda create -n py310 python=3.10
(base) $ conda activate py310
```
* Then, you can run tests using tox
```console
(py310) $ tox run -e pre-merge-all-py310
```
* Or set up dev environment
```console
(py310) $ tox devenv -e pre-merge-all-py310 .venv
```




### Checklist




### License

      Feel free to contact the maintainers if that's a concern.

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```
